tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,676 @@
+			testUserAlias + " " +
+			"-propagate=false -role CNS-DATASTORE-" + testUser + " '" + datastores[i] + "'"
+		framework.Logf("Assign datastore level Permissions to user: %s ", setPermissionForDataStore)
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForDataStore)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForDataStore, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setSearchlevelPermission method is used to set search level permissions
+func setSearchlevelPermission(vcAddress string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUserAlias string, testUser string) error {
+	setSearchLevelPermission := govcLoginCmd(vcAddress) + "govc permissions.set -principal " + testUserAlias +
+		" -propagate=false -role CNS-SEARCH-AND-SPBM-" + testUser + " /"
+	framework.Logf("Assign search level Permissions to user: %s ", setSearchLevelPermission)
+	result, err := sshExec(sshClientConfig, masterIp, setSearchLevelPermission)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			setSearchLevelPermission, masterIp, err)
+	}
+	return nil
+}
+
+// createCsiVsphereConfFile method is used to create csi vsphere conf file
+func createCsiVsphereConfFile(clusterName string, clusterDistribution string, vCenterIP string,
+	inSecureFlag bool, testUser string, password string, vCenterPort string,
+	dataCenter string, sshClientConfig *ssh.ClientConfig,
+	masterIp string) error {
+	conf := `tee csi-vsphere.conf >/dev/null <<EOF
+[Global]
+cluster-id = "%s"
+cluster-distribution = "%s"
+[VirtualCenter "%s"]
+insecure-flag = "%t"
+user = "%s"
+password = "%s"
+port = "%s"
+datacenters = "%s"`
+
+	createConf := fmt.Sprintf(conf, clusterName, clusterDistribution, vCenterIP, inSecureFlag, testUser, password,
+		vCenterPort, dataCenter)
+	framework.Logf("vsphere config secret file: %s ", createConf)
+	result, err := sshExec(sshClientConfig, masterIp, createConf)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			conf, masterIp, err)
+	}
+	applyConf := "kubectl create secret generic vsphere-config-secret --from-file=csi-vsphere.conf " +
+		"-n vmware-system-csi"
+	framework.Logf("Create vsphere config secret: %s ", applyConf)
+	result, err = sshExec(sshClientConfig, masterIp, applyConf)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			applyConf, masterIp, err)
+	}
+	return nil
+}
+
+// deleteCsiVsphereConfFile method is used to delete csi vsphere conf file
+func deleteCsiVsphereConfFile(sshClientConfig *ssh.ClientConfig, masterIp string) error {
+	deleteConf := "kubectl delete secret vsphere-config-secret --namespace=vmware-system-csi"
```

Comment:
Can you please use APIs to delete secret instead of using kubectl

Reply:
done